### DISCUSSION TITLE
Raspberry Pi 4 starting in --monitor mode

### DISCUSSION THREAD

**Discussion by clriis** [abraunegg/onedrive#1902] [Q&A] [answered]
Hi and thanks for an amazing piece of work.

I've installed onedrive on a headless RPi 4 (64 bit Raspberry OS) using Debian 11 installation guide and it works fine but defaults to --monitor mode  after installation which I don't want.

I don't see any symlink in /etc/systemd/user

I've disabled onedrive@.service and confirmed it is dead

But still onedrive --monitor starts up a process on system start.

I've checked init.d and even crontab and rc.local.

I can kill the process and it seems to remain dead. but I would like everything to work properly.

Please help!

[config.txt](https://github.com/abraunegg/onedrive/files/8365638/config.txt)

[claus.onedrive.log](https://github.com/abraunegg/onedrive/files/8365611/claus.onedrive.log)

![onedrive_process](https://user-images.githubusercontent.com/16999072/160470262-1be23da8-17e0-4071-a129-ee5f4888ed5a.png)
![sudo systemctl list-unit-files --type=service](https://user-images.githubusercontent.com/16999072/160470267-c20f69c6-a7e0-4421-bf2d-d1ddfd06d367.png)
![systemctl status](https://user-images.githubusercontent.com/16999072/160470269-24e163d6-d119-4c30-8090-a56143d9fcf3.png)

**Comment by abraunegg** [comment, marked as answer]
@clriis 

> I've installed onedrive on a headless RPi 4 using Debian 11 installation guide and it works fine but defaults to --monitor mode after installation which I don't want.
> 
> I don't see any symlink in /etc/systemd/user
> 
> I've disabled [onedrive@.service](mailto:onedrive@.service) and confirmed it is dead
> 
> But still onedrive --monitor starts up a process on system start.
> 
> I've checked init.d and even crontab and bash.

The issue will be with the package you have installed. It *will* have installed and enabled a systemd service *somewhere*. This is *why* it is starting up on system start - no other reason.

On my Debian 11 test system (clean install), below is the output when installing the client:

**First before any install:**
Lets check the systemd directory for 'onedrive' entries ..
```
alex@debian11:~$ sudo find /etc/systemd/ | grep onedrive
[sudo] password for alex: 
alex@debian11:~$ 
```

OK .. no entries .. install the 'onedrive' client from the OpenSuSE Build Service:
```
Linux debian11 5.10.0-12-amd64 #1 SMP Debian 5.10.103-1 (2022-03-07) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Tue Mar 29 06:42:21 2022 from X.X.X.X
alex@debian11:~$ echo 'deb https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_11/ ./' | sudo tee /etc/apt/sources.list.d/onedrive.list
[sudo] password for alex: 
deb https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_11/ ./
alex@debian11:~$ wget -qO - https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_11/Release.key | sudo apt-key add -
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
OK
alex@debian11:~$ sudo apt-get update
Hit:1 http://security.debian.org/debian-security bullseye-security InRelease
Hit:2 http://deb.debian.org/debian bullseye InRelease
Hit:3 http://deb.debian.org/debian bullseye-updates InRelease
Get:4 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_11 ./ InRelease [1,596 B]
Get:5 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_11 ./ Packages [1,590 B]
Fetched 3,186 B in 2s (1,524 B/s)
Reading package lists... Done
alex@debian11:~$ sudo apt install onedrive
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libphobos2-ldc-shared94
The following NEW packages will be installed:
  libphobos2-ldc-shared94 onedrive
0 upgraded, 2 newly installed, 0 to remove and 57 not upgraded.
Need to get 1,886 kB of archives.
After this operation, 8,704 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian bullseye/main amd64 libphobos2-ldc-shared94 amd64 1:1.24.0-2 [1,280 kB]
Get:2 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_11 ./ onedrive 2.4.16-1~np1 [606 kB]
Fetched 1,886 kB in 7s (255 kB/s)                                                                                                                                                            
Selecting previously unselected package libphobos2-ldc-shared94:amd64.
(Reading database ... 139945 files and directories currently installed.)
Preparing to unpack .../libphobos2-ldc-shared94_1%3a1.24.0-2_amd64.deb ...
Unpacking libphobos2-ldc-shared94:amd64 (1:1.24.0-2) ...
Selecting previously unselected package onedrive.
Preparing to unpack .../onedrive_2.4.16-1~np1_amd64.deb ...
Unpacking onedrive (2.4.16-1~np1) ...
Setting up libphobos2-ldc-shared94:amd64 (1:1.24.0-2) ...
Setting up onedrive (2.4.16-1~np1) ...
Created symlink /etc/systemd/user/default.target.wants/onedrive.service → /usr/lib/systemd/user/onedrive.service.    <--- THIS IS CREATED IN ERROR ..........
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for libc-bin (2.31-13+deb11u2) ...
alex@debian11:~$

```

During the install, you can see the created systemd entry ... but lets check:

```
alex@debian11:~$ sudo find /etc/systemd/ | grep onedrive
/etc/systemd/user/default.target.wants/onedrive.service
alex@debian11:~$ 

```

On system reboot - this is *why* the service is starting. This is also clearly detailed here: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages

OK .. so what if you did not use the OpenSuSE Build Service ... and used the Debian 11 Universe Package?

```
alex@debian11:~$ sudo apt install onedrive
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libphobos2-ldc-shared94
The following NEW packages will be installed:
  libphobos2-ldc-shared94 onedrive
0 upgraded, 2 newly installed, 0 to remove and 57 not upgraded.
Need to get 1,800 kB of archives.
After this operation, 8,389 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian bullseye/main amd64 libphobos2-ldc-shared94 amd64 1:1.24.0-2 [1,280 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 onedrive amd64 2.4.10-1 [521 kB]
Fetched 1,800 kB in 0s (3,670 kB/s)
Selecting previously unselected package libphobos2-ldc-shared94:amd64.
(Reading database ... 139945 files and directories currently installed.)
Preparing to unpack .../libphobos2-ldc-shared94_1%3a1.24.0-2_amd64.deb ...
Unpacking libphobos2-ldc-shared94:amd64 (1:1.24.0-2) ...
Selecting previously unselected package onedrive.
Preparing to unpack .../onedrive_2.4.10-1_amd64.deb ...
Unpacking onedrive (2.4.10-1) ...
Setting up libphobos2-ldc-shared94:amd64 (1:1.24.0-2) ...
Setting up onedrive (2.4.10-1) ...
Created symlink /etc/systemd/user/default.target.wants/onedrive.service → /usr/lib/systemd/user/onedrive.service. <--- THIS IS CREATED IN ERROR ..........
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for libc-bin (2.31-13+deb11u2) ...
alex@debian11:~$ 
```

So 2 issues:
1. Obsolete client version, unsupported
2. systemd entry created in error ....



Please can you check your systemd directory for the offending entry and remove it as per https://github.com/abraunegg/onedrive/issues/1274#issuecomment-813846904

```
sudo rm /etc/systemd/user/default.target.wants/onedrive.service
```

**Reply by clriis** [reply]
Sorry. I was really convinced I had checked for the symlink, but it was there. Everything is fine now after I deleted it.

My onedrive version is: v2.4.16-1~np1. Is that correct?

Thanks a lot

**Reply by abraunegg** [reply]
> My onedrive version is: v2.4.16-1~np1. Is that correct?

Yes this is the correct version to be running - anything else (less than 2.4.16) is obsolete and unsupported.
